touchforms/formplayer/api.py

```diff
@@ -287,8 +288,23 @@ def post_data(data, auth=None, content_type="application/json"):
     if domain:
         d['uses_sql_backend'] = use_sqlite_backend(domain)
     # just default to old server for now
-    url = settings.XFORMS_PLAYER_URL + '/' + d['action']
-    return post_data_helper(d, auth, content_type, url)
+    return perform_experiment(d, auth, content_type)
+
+
+def perform_experiment(d, auth, content_type):
+    experiment = FormplayerExperiment(name=d["action"], context={'request': d})
+    with experiment.control() as c:
+        c.record(post_data_helper(d, auth, content_type, settings.XFORMS_PLAYER_URL))
+    with experiment.candidate() as c:
+        # If we should already have a session, look up its ID in the experiment mapping. it better be there.
+        # This is terrible, but we use both in different places.
+        if "session_id" in d:
+            d["session_id"] = FormplayerExperiment.session_id_mapping.get(d["session_id"])
+        if "session-id" in d:
+            d["session-id"] = FormplayerExperiment.session_id_mapping.get(d["session-id"])
+        c.record(post_data_helper(d, auth, content_type, settings.FORMPLAYER_URL + "/" + d["action"]))
+    objects = experiment.run()
```

Comment:
when this gets run, are there side effects from running it in both places? for example, do two forms get submitted, one from each place, if someone gets all the way through a survey?

Reply:
Good question. In the `prompt` mode Formplayer will never actually submit a form as currently configured - the output XML is returned to the calling service when the form reaches the end and will be ignored by the experiment. I assume SMS doesn't support case claim? That's the only other side effect I could think of (besides the creation of Formplayer sessions and user DBs which shouldn't impact anything).